cmd/medsky/bgs/bgs.go

```diff
@@ -0,0 +1,1440 @@
+	slOpts.MaxQueuePerPDS = config.MaxQueuePerPDS
+	slOpts.Logger = bgs.log
+	s, err := NewSlurper(db, bgs.handleFedEvent, slOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	bgs.slurper = s
+
+	if err := bgs.slurper.RestartAll(); err != nil {
+		return nil, err
+	}
+
+	bgs.nextCrawlers = config.NextCrawlers
+	bgs.httpClient.Timeout = time.Second * 5
+
+	return bgs, nil
+}
+
+func (bgs *BGS) StartMetrics(listen string) error {
+	http.Handle("/metrics", promhttp.Handler())
+	return http.ListenAndServe(listen, nil)
+}
+
+func (bgs *BGS) Start(addr string) error {
+	var lc net.ListenConfig
+	ctx, cancel := context.WithTimeout(context.Background(), serverListenerBootTimeout)
+	defer cancel()
+
+	li, err := lc.Listen(ctx, "tcp", addr)
+	if err != nil {
+		return err
+	}
+	return bgs.StartWithListener(li)
+}
+
+func (bgs *BGS) StartWithListener(listen net.Listener) error {
+	e := echo.New()
+	e.HideBanner = true
+
+	e.Use(middleware.CORSWithConfig(middleware.CORSConfig{
+		AllowOrigins: []string{"*"},
+		AllowHeaders: []string{echo.HeaderOrigin, echo.HeaderContentType, echo.HeaderAccept, echo.HeaderAuthorization},
+	}))
+
+	if !bgs.ssl {
+		e.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
+			Format: "method=${method}, uri=${uri}, status=${status} latency=${latency_human}\n",
+		}))
+	} else {
+		e.Use(middleware.LoggerWithConfig(middleware.DefaultLoggerConfig))
+	}
+
+	// React uses a virtual router, so we need to serve the index.html for all
+	// routes that aren't otherwise handled or in the /assets directory.
+	e.File("/dash", "public/index.html")
+	e.File("/dash/*", "public/index.html")
+	e.Static("/assets", "public/assets")
+
+	e.Use(MetricsMiddleware)
+
+	e.HTTPErrorHandler = func(err error, ctx echo.Context) {
+		switch err := err.(type) {
+		case *echo.HTTPError:
+			if err2 := ctx.JSON(err.Code, map[string]any{
+				"error": err.Message,
+			}); err2 != nil {
+				bgs.log.Error("Failed to write http error", "err", err2)
+			}
+		default:
+			sendHeader := true
+			if ctx.Path() == "/xrpc/com.atproto.sync.subscribeRepos" {
+				sendHeader = false
+			}
+
+			bgs.log.Warn("HANDLER ERROR: (%s) %s", ctx.Path(), err)
+
+			if strings.HasPrefix(ctx.Path(), "/admin/") {
+				ctx.JSON(500, map[string]any{
+					"error": err.Error(),
+				})
+				return
+			}
+
+			if sendHeader {
+				ctx.Response().WriteHeader(500)
+			}
+		}
+	}
+
+	// TODO: this API is temporary until we formalize what we want here
+
+	e.GET("/xrpc/com.atproto.sync.subscribeRepos", bgs.EventsHandler)
+	e.GET("/xrpc/com.atproto.sync.requestCrawl", bgs.HandleComAtprotoSyncRequestCrawl)
+	e.POST("/xrpc/com.atproto.sync.requestCrawl", bgs.HandleComAtprotoSyncRequestCrawl)
+	e.GET("/xrpc/com.atproto.sync.listRepos", bgs.HandleComAtprotoSyncListRepos)
+	e.GET("/xrpc/com.atproto.sync.getLatestCommit", bgs.HandleComAtprotoSyncGetLatestCommit)
+	e.GET("/xrpc/_health", bgs.HandleHealthCheck)
+	e.GET("/_health", bgs.HandleHealthCheck)
+	e.GET("/", bgs.HandleHomeMessage)
+
```

Comment:
doesn't have to be in this PR / iteration, but we'll want to have `getRepo` as an endpoint on the relay, which to start does an HTTP redirect to upstream (eg, PDS)

Reply:
added 302 handler for getRepo